contracts/resolvers/DelegatableResolver.sol

```diff
@@ -0,0 +1,121 @@
+pragma solidity >=0.8.4;
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import "./profiles/ABIResolver.sol";
+import "./profiles/AddrResolver.sol";
+import "./profiles/ContentHashResolver.sol";
+import "./profiles/DNSResolver.sol";
+import "./profiles/InterfaceResolver.sol";
+import "./profiles/NameResolver.sol";
+import "./profiles/PubkeyResolver.sol";
+import "./profiles/TextResolver.sol";
+import "./profiles/ExtendedResolver.sol";
+import "./Multicallable.sol";
+import "./IDelegatableResolver.sol";
+
+/**
+ * A delegated resolver that allows the resolver owner to add an operator to update records of a node on behalf of the owner.
+ * address.
+ */
+contract DelegatableResolver is
+    Ownable,
+    Multicallable,
+    ABIResolver,
+    AddrResolver,
+    ContentHashResolver,
+    DNSResolver,
+    InterfaceResolver,
+    NameResolver,
+    PubkeyResolver,
+    TextResolver,
+    ExtendedResolver
+{
+    using BytesUtils for bytes;
+
+    // Logged when an operator is added or removed.
+    event Approval(
+        bytes32 indexed node,
+        address indexed operator,
+        bytes name,
+        bool approved
+    );
+
+    error NotAuthorized(bytes32 node);
+
+    constructor(address owner) {
+        operators[bytes32(0)][owner] = true;
+        transferOwnership(owner);
+    }
+
+    //node => (delegate => isAuthorised)
+    mapping(bytes32 => mapping(address => bool)) operators;
+
+    /**
+     * @dev Check to see if the operator has been approved by the owner for the node.
+     */
+    function getAuthorizedNode(
+        bytes memory name,
+        uint256 offset,
+        address operator
+    ) public view returns (bytes32 node, bool authorized) {
+        uint256 len = name.readUint8(offset);
+        node = bytes32(0);
+        if (len > 0) {
```

Comment:
Doing the check for the current node before recursing would save gas.

Reply:
Not sure what you mean. `if (len > 0) ` at line 62 is happening before recursive call happens on line 64.

You mean flipping the if order like this?

```
        if(len <= 0){
            return (node, authorized || operators[node][operator]);
        } else {
            bytes32 label = name.keccak(offset + 1, len);
            (node, authorized) = getAuthorizedNode(
                name,
                offset + len + 1,
                operator
            );
            node = keccak256(abi.encodePacked(node, label));
        }
        
```